django/db/backends/base/schema.py

```diff
@@ -992,12 +997,19 @@ def _index_include_sql(self, model, columns):
 
     def _create_index_sql(self, model, *, fields=None, name=None, suffix='', using='',
                           db_tablespace=None, col_suffixes=(), sql=None, opclasses=(),
-                          condition=None, include=None):
+                          condition=None, include=None, expressions=None):
         """
-        Return the SQL statement to create the index for one or several fields.
+        Return the SQL statement to create the index for one or several fields or expressions.
         `sql` can be specified if the syntax differs from the standard (GIS
         indexes, ...).
         """
+        fields = fields or []
+        expressions = expressions or []
+        compiler = self.connection.ops.compiler('SQLCompiler')(
+            query=Query(model, alias_cols=False),
+            connection=self.connection,
+            using=self.connection.alias
+        )
```

Comment:
I wonder if this is not a bit cleaner:

```suggestion
        compiler = Query(model, alias_cols=False).get_compiler(connection=self.connection)
```

Reply:
Yes, much better, thanks.